### PR TITLE
periodically calls the tts module when a paragraph has been generated

### PR DESCRIPTION
**Pull request by CaseyHaralson**
When a character has text being generated, a timer starts and passes each paragraph to the tts module. This allows the tts module to start returning speech faster and helps with long speech generation breaking the tts module (some endpoints have a 2000 char text limit).

This might need a configuration option, but I'm not sure.

### CODE REVIEW

**Comment by Cohee1207** [comment]
This should be an option.

**Comment by CaseyHaralson** [comment]
Option added.

File: public/scripts/extensions/tts/index.js [hunk 1]
```diff
@@ -779,6 +791,70 @@ async function onMessageDeleted() {
...
+
+    // the last message was from the user
+    if (context.chat[lastMessageId].is_user) {
+      return;
+    }
+
+    const lastMessage = structuredClone(context.chat[lastMessageId]);
+    const lastMessageText = lastMessage?.mes ?? '';
+
+    // look for double ending lines which should indicate the end of a paragraph
```
Comment (Cohee1207) [inline public/scripts/extensions/tts/index.js:848]: Not all LLMs use double newlines as paragraph boundaries. Some use one.

File: public/scripts/extensions/tts/index.js [hunk 2]
```diff
@@ -1008,6 +1084,10 @@ $(document).ready(function () {
                             <input type="checkbox" id="tts_auto_generation">
                             <small>Auto Generation</small>
                         </label>
+                        <label class="checkbox_label" for="tts_periodic_auto_generation">
+                            <input type="checkbox" id="tts_periodic_auto_generation">
+                            <small>Periodically auto generate</small>
```
Comment (Cohee1207) [inline public/scripts/extensions/tts/index.js]: `Narrate by paragraphs (streaming)` could be a more accurate label

**Comment by Cohee1207** [comment]
I gave it a try, and it works great, albeit with some gotchas:

1. Auto-generate should be enabled, but this requirement is straightforward and obvious.
2. If "Narrate only the translated text" is enabled, the generation will be done only after completing the stream when the auto-translate kicks in (if enabled of course).

This should be good to merge into staging now, for collecting further feedback before releasing.

**Review by Cohee1207 — APPROVED** [review]

**Pull request merged**